sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/NotificationHook/WebNotificationHook.cs

```diff
@@ -17,72 +16,65 @@ public partial class WebNotificationHook : NotificationHook
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="WebNotificationHook"/> class.
-        /// <param name="name">The name to assign to the hook.</param>
-        /// <param name="endpoint">The API address to be called when an alert is triggered.</param>
-        /// <exception cref="ArgumentNullException"><paramref name="name"/> or <paramref name="endpoint"/> is null.</exception>
-        /// <exception cref="ArgumentException"><paramref name="name"/> or <paramref name="endpoint"/> is empty.</exception>
         /// </summary>
-        public WebNotificationHook(string name, string endpoint)
-            : base(name)
+        public WebNotificationHook()
         {
-            Argument.AssertNotNullOrEmpty(endpoint, nameof(endpoint));
-
-            HookParameter = new WebhookHookParameter(endpoint, default, default, new ChangeTrackingDictionary<string, string>(), default, default);
```

Comment:
We need to pass an internal generated model called `HookParameter` to be serialized when sending a request to the service. Originally, we had an internal `HookParameter` and we were modifying it in every property setter call. However, to create a `HookParameter` we need a non-null `endpoint`, which we don't have because it's not required by the constructor anymore.

Now we don't store an internal `HookParameter` object anymore. We simply build a new one once we need to serialize it.